lib/ansible/modules/source_control/git.py

```diff
@@ -1240,8 +1252,9 @@ def main():
                 result.update(changed=True, after=remote_head)
                 module.exit_json(**result)
 
-            # Switch to version specified
-            submodule_update(git_path, module, dest, track_submodules, force=force)
+    if (submodules_updated or update_submodules) and not bare:
```

Comment:
@samdoran This if clause preserves current behaviour. If i change update_submodules=yes by default it will change current behaviour and some integration tests need to be fixed.